src/controllers/profile.js

```diff
@@ -16,28 +16,40 @@ exports.updateProfile = async (req, res, next) => {
       .json({ success: false, errors: validationErrors.array() });
   }
 
-  let hashedPassword;
+  let user = await User.findById(req.user._id);
 
-  if (req.body.password) {
-    hashedPassword = await bcrypt.hash(req.body.password, 10);
+  if (!user) {
+    return new ErrorResponse('user does not exists!', 400);
   }
   const userexist = await User.findOne({ email: req.body.email });
   if (userexist) {
     return next(new ErrorResponse('Email already exists!', 400));
   }
-  const user = await User.findByIdAndUpdate(
+
+  if (req.body.password) {
+    if (!(await bcrypt.compare(req.body.oldpassword, user.password))) {
+      return next(
+        new ErrorResponse(
+          'Your old password is not correct please enter again!',
+          401
+        )
+      );
+    }
+    console.log(`l${req.body.password}`);
+    req.body.password = await bcrypt.hash(req.body.password, 10);
+  }
+
+  user = await User.findByIdAndUpdate(
     req.user._id,
     {
       $set: {
         firstName: req.body.firstName,
         lastName: req.body.lastName,
         email: req.body.email,
         schoolName: req.body.schoolName,
-        password: hashedPassword,
+        password: req.body.password,
         phoneNumber: req.body.phoneNumber,
         address: req.body.address,
-        provider: req.body.provider,
-        providerId: req.body.providerId,
```

Comment:
So since these fields are all updatable, all of them need to be included in `swagger` documentation. Please check docs for the API endpoint and make sure they're 100% correct.